### DISCUSSION TITLE
Manually view OTP token

### DISCUSSION THREAD

**Discussion by ronildo** [supabase/supabase#27439] [Questions] [answered]
How can I manually view an OTP token?
On auth.one_time_tokens I can see a token_hash, but I don't know how to decode it 😳

I have some users complaining they are not getting their token and I would like to manually send it to them when needed.

**Comment by silentworks** [comment, marked as answer]
It's available to your email templates using the `{{ .Token }}` template tag. If you need to manually send one to the user you can use `supabase.auth.admin.generateLink()` which will return the token. https://supabase.com/docs/reference/javascript/auth-admin-generatelink

**Reply by ronildo** [reply]
Thank you so much!
